### PR TITLE
Replace uses of some string semantics with constant string literals

### PR DESCRIPTION
**Pull request by zoecarver**
This patch updates all uses of `string.equals` and `string.makeUTF8` to use a constant string literal. I have another patch to do this on a more broad scale (coming soon).

### CODE REVIEW

**Comment by zoecarver** [comment]
cc @gottesmm

File: include/swift/Strings.h [hunk 1]
```diff
@@ -121,6 +121,12 @@ constexpr static BuiltinNameStringLiteral BUILTIN_TYPE_NAME_SILTOKEN = {
 constexpr static BuiltinNameStringLiteral BUILTIN_TYPE_NAME_WORD = {
     "Builtin.Word"};
 
+/// The name of the semantics of make utf8 string
```
Comment (gottesmm) [inline include/swift/Strings.h]: The string passed as a semantics attribute to signal to the optimizer that a function is a call functionally to string.makeUTF8?

File: include/swift/Strings.h [hunk 2]
```diff
@@ -121,6 +121,12 @@ constexpr static BuiltinNameStringLiteral BUILTIN_TYPE_NAME_SILTOKEN = {
...
+constexpr static StringLiteral STRING_MAKE_UTF8 = "string.makeUTF8";
+
+/// The name of the semantics of string equal
```
Comment (gottesmm) [inline include/swift/Strings.h]: Something similar as the above?

File: include/swift/Strings.h (hunk 1 above)
Comment (zoecarver) [reply include/swift/Strings.h]: That's better :)

**Comment by slavapestov** [comment]
Is this obsoleted by https://github.com/apple/swift/pull/28037?

**Comment by zoecarver** [comment]
Yep. I was hoping that this could land quickly so that I could also land #27909 but, it seems that #28037 will land first.

**Pull request closed without merging**